File: days/app/Http/Resources/Assessment.php
<?php

namespace App\Http\Resources;

use Illuminate\Http\Resources\Json\JsonResource;

/** @mixin \App\Models\Assessment */
class Assessment extends JsonResource
{
    /**
     * @param \Illuminate\Http\Request $request
     * @return array
     */
    public function toArray($request)
    {
        return [
            'id' => $this->id,
            'subject' => $this->subject->name,
            'contents' => $this->contents,
            'assessment_date' => $this->assessment_date,
        ];
    }
}

File: days_workers/lib/models/Assessment.dart
class Assessment {
  int id;
  String subject;
  String contents;
  String assessmentDate;

  Assessment({this.id, this.subject, this.contents, this.assessmentDate});

  Assessment.fromJson(Map<String, dynamic> json) {
    id = json['id'];
    subject = json['subject'];
    contents = json['contents'];
    assessmentDate = json['assessment_date'];
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['id'] = this.id;
    data['subject'] = this.subject;
    data['contents'] = this.contents;
    data['assessment_date'] = this.assessmentDate;
    return data;
  }
}
